### PR TITLE
Add an assert in case of logic error and replace value copy with cons…

### PR DESCRIPTION
**Pull request by herve-er**
QStrings are currently passed by value where they could be passed by (const) references.

### CODE REVIEW

File: src/server/socketlistener.cpp [hunk 1]
```diff
@@ -41,6 +41,7 @@ SocketListener::SocketListener(QIODevice *socket) : socket(socket) {
 void SocketListener::sendMessage(const QString &message, bool doWait) const {
     if (_threadId != std::this_thread::get_id()) {
         LOGW_ERROR(KDC::Log::instance()->getLogger(), L"SocketListener::sendMessage should only be called from the main thread");
```
Comment (luc-guyot-infomaniak) [inline src/server/socketlistener.cpp:43]: Unless you absolutely want to log an error message, you can use a one line assertion with a message as depicted in this SO post: https://stackoverflow.com/questions/3692954/add-custom-messages-in-assert.

Comment (herve-er) [reply src/server/socketlistener.cpp]: No issue for me to remove this log line: [627464b](https://github.com/Infomaniak/desktop-kDrive/pull/238/commits/627464bb363dc2acb289bd7944b302b5e2398445)

File: src/server/socketlistener.cpp [hunk 2]
```diff
@@ -40,7 +40,7 @@ SocketListener::SocketListener(QIODevice *socket) : socket(socket) {
 
 void SocketListener::sendMessage(const QString &message, bool doWait) const {
     if (_threadId != std::this_thread::get_id()) {
-        LOGW_ERROR(KDC::Log::instance()->getLogger(), L"SocketListener::sendMessage should only be called from the main thread");
+        assert(false && "SocketListener::sendMessage should only be called from the main thread");
```
Comment (luc-guyot-infomaniak) [inline src/server/socketlistener.cpp:43]:
```suggestion
        assert(_threadId != std::this_thread::get_id() && "SocketListener::sendMessage should only be called from the main thread");
```

File: src/server/socketlistener.cpp [hunk 3]
```diff
@@ -40,7 +40,7 @@ SocketListener::SocketListener(QIODevice *socket) : socket(socket) {
 
 void SocketListener::sendMessage(const QString &message, bool doWait) const {
     if (_threadId != std::this_thread::get_id()) {
```
Comment (herve-er) [inline src/server/socketlistener.cpp:42]:
```suggestion
```

File: src/server/socketlistener.cpp [hunk 4]
```diff
@@ -40,7 +40,7 @@ SocketListener::SocketListener(QIODevice *socket) : socket(socket) {
...
-        LOGW_ERROR(KDC::Log::instance()->getLogger(), L"SocketListener::sendMessage should only be called from the main thread");
+        assert(_threadId != std::this_thread::get_id() && "SocketListener::sendMessage should only be called from the main thread");
     }
```
Comment (herve-er) [inline src/server/socketlistener.cpp:44]:
```suggestion
```

File: src/server/socketlistener.cpp [hunk 5]
```diff
@@ -39,9 +39,7 @@ SocketListener::SocketListener(QIODevice *socket) : socket(socket) {
 }
 
 void SocketListener::sendMessage(const QString &message, bool doWait) const {
-    if (_threadId != std::this_thread::get_id()) {
-        LOGW_ERROR(KDC::Log::instance()->getLogger(), L"SocketListener::sendMessage should only be called from the main thread");
-    }
+        assert(_threadId != std::this_thread::get_id() && "SocketListener::sendMessage should only be called from the main thread");
```
Comment (herve-er) [inline src/server/socketlistener.cpp:42]:
```suggestion
    assert(_threadId != std::this_thread::get_id() && "SocketListener::sendMessage should only be called from the main thread");
```

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Pull request merged**